### DISCUSSION TITLE
Airflow 2.x.x :: Task stuck in queued state even though pool slots available

### DISCUSSION THREAD

**Discussion by aa3pankaj** [apache/airflow#19309] [Q&A]
python dependencies:

apache-airflow==2.1.3
redis==3.5.3
celery==4.4.7
(Is combination of redis 3.5.3 and celery 4.4.7 compatible?)

airflow configs:

core.parallelism = 1000
core.dag_concurrency = 400
core.max_active_runs_per_dag = 1
celery.worker_concurrency = 6
celery.broker_url = redis
celery.result_backend = mysql

Facing this issue after upgrade to airflow 2.0

Not all tasks stuck in "queued" state, only 1 task stuck at a time (out of 100+).

Sometimes tasks stuck in "running" state also.

When task stucks with state "queued", this task is missing in redis queue also

This issue happens once in 1-2 days.

This happens even though pool slots available to use (Slots = 128, Running slots = 0, Queued slots = 1)

As per worker logs (for both stuck tasks and normal tasks):
`INFO/MainProcess] Received task: airflow.executors.celery_executor.execute_command[066398d7-98e0-4f17-bd2b-93ae36438577]
INFO/ForkPoolWorker-6] Executing command in Celery: ['airflow', 'tasks', 'run', 'dag_name', 'task01', '2021-10-26T14:00:00+00:00', '--local', '--pool', 'default_pool', '--subdir', '/dag_dir/dag.py']
`

I see below log for all normal working tasks, this is missing for the stuck tasks:
`WARNING/ForkPoolWorker-1] Running <TaskInstance: dag.task02 2021-10-26T14:00:00+00:00 [queued]> on host worker02.localdomain`

@ephraimbuddy Need help here
Currently resolving this issue by manually marking the DagRun to "failed"

**Comment by sdhzlzhk** [comment]
I think your task is failed, you can find the task in flower UI. 
![image](https://user-images.githubusercontent.com/7436812/144342622-ec9f3f62-8b14-47b8-8780-3bd461e8a4e5.png)

**Comment by ephraimbuddy** [comment]
Since you are already in 2.1.3 I will advise you to move to 2.2.2, the stuck task issue has been resolved to some extent in newer versions
